python_modules/dagster/dagster/_core/storage/event_log/sql_event_log.py

```diff
@@ -3245,6 +3245,68 @@ def get_updated_data_version_partitions(
 
         return updated_partitions
 
+    def order_assets_by_last_materialized_time(
+        self, asset_keys: Optional[Sequence[AssetKey]], descending: bool = False
+    ) -> Sequence[AssetKey]:
+        """Given a list of asset keys, returns the keys ordered by the time they most recently had a
+        materialization-related or observation event (based on the last_materialization_timestamp
+        which is updated on planned, materialization, and observation events). Assets that have never
+        been materialized or have been wiped since the latest event are considered the oldest.
+
+        If no list of asset keys is provided, all asset keys in the DB are returned in sorted order.
+        This means that if an asset is not in the DB (ie if it has never been materialized) it will not
+        be returned.
+
+        descending=True - newest asset first, never materialized assets last
+        descending=False (default) - never materialized assets first, then oldest asset
+        """
+        if not self.has_asset_key_index_cols() or not self.has_secondary_index(
+            ASSET_KEY_INDEX_COLS
+        ):
+            # punting on these cases for prototype
+            raise NotImplementedError()
+
+        if descending:
+            order_by = AssetKeyTable.c.last_materialization_timestamp.desc()
+        else:
+            order_by = AssetKeyTable.c.last_materialization_timestamp.asc()
+
+        query = (
+            db_select([AssetKeyTable.c.asset_key])
+            .order_by(order_by)
```

Comment:
i think the key thing is that doing this order by on the backend/in porgres only makes sense if postgres knows more about how to order it or paginate it efficiently than the frontend or caller does, which isn't the case here since we don't have an index